system-design/margo-overview/software-composition.md

```diff
@@ -0,0 +1,285 @@
+        Component(cim1, "Container Image 1", "Archive")
+    }
+
+    Rel(wldc1, ca1, "refers")
+    Rel(cc1, cim1, "refers")
+
+    UpdateElementStyle(ad, $fontColor="white", $bgColor="blue", $borderColor="grey")
+    UpdateElementStyle(ca1, $fontColor="black", $bgColor="lightsalmon", $borderColor="grey")
+```
+
+The application and contained components are typically configurable with the option of providing default values.
+
+### 2. Software Deployment
+
+When a device gets the instruction to run an [Application][application] (over a desired-state specified with an [`ApplicationDeployment` object][deployment-definition]), its [Workload Fleet Management Agent][wfma] interacts with the [providers][provider-model].
+That way all [Workloads][workload] needed for an [Application][application] should get started and the desired state should be reached.
+
+```mermaid
+C4Context
+    title Software Deployment stage
+
+    UpdateLayoutConfig($c4BoundaryInRow="1", $c4ShapeInRow="2")
+
+    Enterprise_Boundary(be, "Backend") {
+        System(apdb, "ApplicationDeployment", "Application Deployment specification")
+        System(appb, "ApplicationDescription", "Application Package")
+        System(cmp, "Component")
+        Rel(appb, apdb, "instantiation")
+        Rel(appb, cmp, "requires")
+    }
+
+    Enterprise_Boundary(dev, "Device") {
+        System(apdd, "ApplicationDeployment", "Application Deployment specification")
+        System(wls, "Workload")
+        Rel(apdd, wls, "configures")
+    }
+
+    BiRel(apdb, apdd, "same")
+    Rel(cmp, wls, "instantiation")
+    UpdateElementStyle(appb, $fontColor="black", $bgColor="blue", $borderColor="grey")
+    UpdateElementStyle(appd, $fontColor="black", $bgColor="blue", $borderColor="grey")
+    UpdateElementStyle(apdb, $fontColor="black", $bgColor="green", $borderColor="grey")
+    UpdateElementStyle(apdd, $fontColor="black", $bgColor="green", $borderColor="grey")
+```
+
+In this stage the [providers][provider-model] are responsible for managing the individual [Workloads][workload].
+
+On a Helm v3 [Deployment Profiles][deployment-profile], a [Workload Fleet Management Agent][wfma] implementation could utilize the Helm API to start the individual Helm Charts.
+
+On a Compose [Deployment Profiles][deployment-profile], a [Workload Fleet Management Agent][wfma] implementation could utilize the Compose CLI to start the individual [Workloads][workload].
+
+The following diagram shows the result of reaching the desired state for an [Application][application] with a Helm v3 [Deployment Profile][deployment-profile] (the result of `helm install`).
+
+```mermaid
+C4Component
+    title Application Deployment: Helm v3 deployment profile
+
+    UpdateLayoutConfig($c4BoundaryInRow="3", $c4ShapeInRow="1")
+
+    System_Boundary(dev1, "Device 1") {
+        System_Boundary(woa1, "Workload Fleet Management Agent") {
+            Component(atb1, "Application Description 1", "ApplicationDescription", "YAML document")
```

Comment:
As mentioned by @phil-abb in [this comment](https://github.com/margo/specification/pull/81#discussion_r2080016984), this should be Application Deployment and not Application Description.

Reply:
@phil-abb could you please check if now it's correct and resolve this conversation, if so? Thanks!